### ISSUE TITLE
playlist linking doesn't include limits/sorts

### ISSUE THREAD

**Issue by simon-weber** [simon-weber/Autoplaylists-for-Google-Music#75]
This can cause incorrect results, eg:
- A: all tracks, limit 10
- B: all tracks, limit 10, sort backwards
- C: include A | B, limit 30

Here, C will include 30 results, even though it should only include 20 (A union B). When building the subrules from the linked playlist, the limits and sorts need to also be applied.

**Comment by simon-weber** [comment]
Limits and sorts are attached to the query, not the where clause, so it doesn't work with the rule combining approach.

**Comment by simon-weber** [comment]
And it doesn't look like lovefield is planning support for subqueries any time soon: https://github.com/google/lovefield/issues/28.

We could always implement this this multiple queries, but combining the results seems like a pain. I'll probably just let this problem sit unless we come up with something better in the future.

**Comment by simon-weber** [comment]
This could be solved by switching autoplaylist linking to use the same system as [static playlist linking](https://github.com/simon-weber/Autoplaylists-for-Google-Music/issues/77).

**Comment by metalgeekstl** [comment]
@simon-weber Any updates on this? If this were fixed I'd pay a lifetime subscription! 
The big issue with limits & sorted not working when referencing playlists is when I want to exclude a playlist. 
If A is ratings=up, limit 100 least often played and B is ratings=up, playlists does not equal A, limit 100 least recently played, playlist B ends up with nothing. 
Hope you find a solution soon!

**Comment by simon-weber** [comment]
Switching to the static linking system should be easy, but I'm a bit concerned about how reliable it'll be. When I've got some time I'll look into making that opt-in so you (and whoever else is interested) can test it out.

**Comment by metalgeekstl** [comment]
@simon-weber Awesome. Thanks.

**Comment by metalgeekstl** [comment]
Also, I think most users would be willing to beta test any feature/fix. You're doing an amazing thing here, so I at least will cut you some slack if it doesn't work perfectly.

**Comment by simon-weber** [comment]
Sorry for the silence on this! I've found myself thinking about this again while working on https://github.com/simon-weber/Autoplaylists-for-Google-Music/issues/97. I've realized that switching autoplaylists to the playlist cache isn't ideal, especially when trying to batch updates.

Consider a chain of playlists like 1 -> 2 -> 3 -> ... -> N, where each includes the previous one. The state of N-1 needs to be correct both on the server and in the cache before N can be correct. So, updating all the playlists in one go is impossible; we'd be forced to update them one at a time _and_ in the correct order.

Fortunately, I think it should be pretty easy to switch from rule combining to just calculating the entire dependent playlist and building an `in` query, similar to how the static playlist linking works. I'm not sure why I didn't think of this before: maybe I was worried about the performance of the large `in` query?

Anywho, I think I can have this fixed shortly. Thanks for your patience!
